virocon/distributions.py

```diff
@@ -1494,3 +1495,143 @@ def _fit_mle(self, sample):
+                )
+
+    #  https://stackoverflow.com/a/53640468
+    @staticmethod
+    def _list_scipy_parameters(distribution):
+        """List parameters for scipy.stats.distribution.
+        # Arguments
+            distribution: a string or scipy.stats distribution object.
+        # Returns
+            A list of distribution parameter strings.
+        """
+        if isinstance(distribution, str):
+            distribution = getattr(sts, distribution)
+        if distribution.shapes:
+            parameters = [name.strip() for name in distribution.shapes.split(",")]
+        else:
+            parameters = []
+        # assert distribution is continous
+        parameters += ["loc", "scale"]
+        return parameters
+
+    def _set_default_parameter_values(self):
+        for par_name in self._param_names:
+            if par_name == "loc":
+                setattr(self, par_name, 0)
+            else:
+                setattr(self, par_name, 1)
+            setattr(self, f"f_{par_name}", None)
+
+    @property
+    def parameters(self):
+        return {par_name: getattr(self, par_name) for par_name in self._param_names}
+
+    def _get_scipy_parameters(self, *args, **kwargs):
+        args_with_default = list(self.parameters.values())
+        for i, arg in enumerate(args):
+            if arg is not None:
+                args_with_default[i] = arg
+
+        for key, arg in kwargs.items():
+            try:
+                idx = self._param_names.index(key)
+            except ValueError:
+                # unknown parameter name
+                assert False
+            args_with_default[idx] = arg
+
+        return args_with_default
+
+    def cdf(self, x, *args, **kwargs):
+        scipy_par = self._get_scipy_parameters(*args, **kwargs)
+        return self.scipy_dist.cdf(x, *scipy_par)
+
+    def icdf(self, prob, *args, **kwargs):
+        scipy_par = self._get_scipy_parameters(*args, **kwargs)
+        return self.scipy_dist.ppf(prob, *scipy_par)
+
+    def pdf(self, x, *args, **kwargs):
+        scipy_par = self._get_scipy_parameters(*args, **kwargs)
+        return self.scipy_dist.pdf(x, *scipy_par)
+
+    def draw_sample(self, n, *args, random_state=None, **kwargs):
+        scipy_par = self._get_scipy_parameters(*args, **kwargs)
+        rvs_size = self._get_rvs_size(n, scipy_par)
+        return self.scipy_dist.rvs(*scipy_par, size=rvs_size, random_state=random_state)
+
+    def _fit_mle(self, sample):
+        p0 = [v for k, v in self.parameters.items() if k != "loc" and k != "scale"]
+        loc0 = self.parameters.get("loc", 0)
+        scale0 = self.parameters.get("scale", 1)
+
+        fparams = {}
+        for par_name in self._param_names:
+            val = getattr(self, f"f_{par_name}")
+            if val is not None:
+                fparams[f"f{par_name}"] = val
+
+        if len(fparams) == len(self.parameters):
+            return  # nothing to do
```

Comment:
Should we raise a Warning or Error in that case? 

It's probably a mistake by the user if they call _fit_mle() but have fixed all parameters.